From patchwork Thu Mar  5 03:34:07 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Yuanshen Cao <alex.caoys@gmail.com>
X-Patchwork-Id: 14454003
Received: from mail-qt1-f178.google.com (mail-qt1-f178.google.com
 [209.85.160.178])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 721AA72621
	for <linux-pm@vger.kernel.org>; Thu,  5 Mar 2026 03:34:13 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.160.178
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1772681654; cv=none;
 b=GVeYfdsbqaqKrOIXHuFTjPyxBYSIx4TVtRIMY03D1sptwuMl5az7GkwFHIN09xFzMrDtHjx1aLVVXpJWZoH2uJjhBFta318mFTt9p4Rg1o0xuL913gReBHwGca3024yxNz9n++qVdHOz8Pwsq6BPeDrnr2toCwwKfO6xL3Ec0CQ=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1772681654; c=relaxed/simple;
	bh=G93cII+3A1u7CvKJrZXIJlpca2wo8XCXaCLCELHybVQ=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=aIRObmsNe2q9n0A+Ndjh9CxgnbDm5K9KupQxht+Y/9bCuPxEu6Pcxe3g+0LSrvdVL+QeQ3hoY5aB7Xu7KgWjifoBVgO+4vkQGx6EBNHp5UR2nXWimr3PyeEZdFlo+9JtH1cQt/NBAJXN4Y/z9dVUAAd9d/LgKlNmssP3kJRIz6Y=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=gmail.com;
 spf=pass smtp.mailfrom=gmail.com;
 dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com
 header.b=gHtc/9li; arc=none smtp.client-ip=209.85.160.178
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=gmail.com
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=gmail.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com
 header.b="gHtc/9li"
Received: by mail-qt1-f178.google.com with SMTP id
 d75a77b69052e-506251815a3so67642911cf.0
        for <linux-pm@vger.kernel.org>; Wed, 04 Mar 2026 19:34:13 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1772681652; x=1773286452;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=GoyJS43DjogCNIlYUyLOl9RH46vOQLUOd3a+06hfz+Q=;
        b=gHtc/9liE0Xz/URCsAUZ/VJmEMYRNykrH0X2KvPxyN+gND8W5aEEgP2fHox/Q/FzGv
         gqqWtahGYLEpP2BmRtqJd15wOxQj7mTWSnNe86F+nxd06KKh1kAi2tk0d7c5RSi+dIAx
         dI4IXsgWvhiV9mYFHcMUvZdI8MUfPnR64bG5UFVRkGbJtn2Y3bm6MzzcxbWfX9cuChSi
         n87L+2x+5GDXWp0xItnNLYzJBg1FTdAUdHa6SDSIQxB0Mgg+ALB9oigI6TnED23sFexu
         VhD+KFiUX80qxJCdtvzL4k+sb9kdq0lO3d+j3V5CIlWqxeLG8UKn8qRNwCDKvJ8fN70r
         ZFcg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1772681652; x=1773286452;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=GoyJS43DjogCNIlYUyLOl9RH46vOQLUOd3a+06hfz+Q=;
        b=I7AnGmt0lLZsP5w0Zx0swio6X9MluKmSrsAm3VngPh2BdDCWaduSZkWh4tTsiNGxRc
         OLAmEdhRh+ocDckAb3KkDTbrqLQEDbSVae0m370gAfEaCKKtWRXidT/Osf+GGuBKjYT0
         TyLUbIPEvculOhEn6m5McyqQqD5bprilej3F8IHTepiainyS1JXTGtT/LpsRmEAUdovq
         7ZtvyIGXOoQ04Qk+68qxw+PY1kyqi113bBYE/F61kBR3JlQZkiN87IDKOryLxPR0k/YZ
         wZJF5e4H0OyoGXiGzzktA43EQucdz3so92svqR7V4tP6C6bwB62pkzblXKGI4qwwj4uV
         JA3w==
X-Forwarded-Encrypted: i=1;
 AJvYcCVyQo7Z887KPFtjMncub1/ttIFmNLYCDgd0Gp4h6qNaROUIysmTP+VgZbfGYj8MAj4tmYWjJ7NqrA==@vger.kernel.org
X-Gm-Message-State: AOJu0Yy0stttgeljaRSYWO5/JB9UviLI8F06NO4mpPDsznDJ1UeU3vRy
	XtgDPvkqdxLVc0+ImyrAJW/RCtvAzztD7kUl0v8GyafIlKJDh8eVOsRJvxmgwXe8
X-Gm-Gg: ATEYQzzCoq+6p68RcWoZgkq1bWjXIFaZEJfyxqNvTOp5UbqbaimJmsKBYgHCo2B07iA
	siPHk+nt+oF6EpMweSw7vEz9bdveo6zYRTraAOVw5Vo4HJBlpIlUATfMVXpEJ4j77edwIxgkB5x
	QWUZiKOpz+eGjsp/K7X/cKPxvQepFwy4Gro4uVudF1bTPxM7wukVM+2UWEQB4f80Nle0LuMaf2i
	NBlIrykAPvmx/XPTmbQwKGCwWqDIt4mbN8RyZpFyUPDX8NmkCgPHZ5jAzqMyjphABfz9QxjU8MB
	Se2OffM9JjxZI4EQtcvj/yZ/mABFdLoP6Y/QwTdc/25ZgE6nxfM7uyuyZ0S0wzpt/9GZEgpVPuv
	+q4RhGiWXGUUnyFXlrFOImPT7eVqGZWfNfto1OWvwHCqV0G1S+fvVLUfH2e5nWF0zZ7sHdHGgAq
	0sekZXInyKk9hfrpFbJKs4y0ic+Xs=
X-Received: by 2002:ac8:5a82:0:b0:502:9e93:617d with SMTP id
 d75a77b69052e-508db36395dmr51002631cf.46.1772681652477;
        Wed, 04 Mar 2026 19:34:12 -0800 (PST)
Received: from [172.17.0.2] ([134.128.219.200])
        by smtp.gmail.com with ESMTPSA id
 d75a77b69052e-507449630b6sm190667601cf.7.2026.03.04.19.34.11
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 04 Mar 2026 19:34:12 -0800 (PST)
From: Yuanshen Cao <alex.caoys@gmail.com>
Date: Thu, 05 Mar 2026 03:34:07 +0000
Subject: [PATCH v2 1/2] dt-bindings: power: Add Support for Allwinner A733
 PCK600 Power Domain Controller
Precedence: bulk
X-Mailing-List: linux-pm@vger.kernel.org
List-Id: <linux-pm.vger.kernel.org>
List-Subscribe: <mailto:linux-pm+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-pm+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260305-b4-pck600-a733-v2-1-ba6bbed7d253@gmail.com>
References: <20260305-b4-pck600-a733-v2-0-ba6bbed7d253@gmail.com>
In-Reply-To: <20260305-b4-pck600-a733-v2-0-ba6bbed7d253@gmail.com>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>, Chen-Yu Tsai <wens@kernel.org>,
 Jernej Skrabec <jernej.skrabec@gmail.com>,
 Samuel Holland <samuel@sholland.org>, Ulf Hansson <ulf.hansson@linaro.org>
Cc: devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-sunxi@lists.linux.dev, linux-kernel@vger.kernel.org,
 linux-pm@vger.kernel.org, Yuanshen Cao <alex.caoys@gmail.com>,
 Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
X-Mailer: b4 0.14.3

The A733 PCK600, similar to A523 PCK600, is likely a customized version
of ARM PCK-600 power controller. They share the same BSP drivers in the
package provided by Radxa, with the only difference being the lack of
resets.

Therefore, document A733 compatible and make resets required only for
the other models, as well as prepare the PD definitions for future
device trees.

Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
Signed-off-by: Yuanshen Cao <alex.caoys@gmail.com>
---
 .../bindings/power/allwinner,sun20i-d1-ppu.yaml        | 17 ++++++++++++++++-
 .../dt-bindings/power/allwinner,sun60i-a733-pck-600.h  | 18 ++++++++++++++++++
 2 files changed, 34 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/power/allwinner,sun20i-d1-ppu.yaml b/Documentation/devicetree/bindings/power/allwinner,sun20i-d1-ppu.yaml
index a28e75a9cb6a..b97361ce2a00 100644
--- a/Documentation/devicetree/bindings/power/allwinner,sun20i-d1-ppu.yaml
+++ b/Documentation/devicetree/bindings/power/allwinner,sun20i-d1-ppu.yaml
@@ -20,6 +20,7 @@ properties:
       - allwinner,sun20i-d1-ppu
       - allwinner,sun55i-a523-pck-600
       - allwinner,sun55i-a523-ppu
+      - allwinner,sun60i-a733-pck-600
 
   reg:
     maxItems: 1
@@ -38,9 +39,23 @@ required:
   - compatible
   - reg
   - clocks
-  - resets
   - '#power-domain-cells'
 
+allOf:
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - allwinner,sun8i-v853-ppu
+              - allwinner,sun20i-d1-ppu
+              - allwinner,sun55i-a523-pck-600
+              - allwinner,sun55i-a523-ppu
+
+    then:
+      required:
+        - resets
+
 additionalProperties: false
 
 examples:
diff --git a/include/dt-bindings/power/allwinner,sun60i-a733-pck-600.h b/include/dt-bindings/power/allwinner,sun60i-a733-pck-600.h
new file mode 100644
index 000000000000..cf476a005b55
--- /dev/null
+++ b/include/dt-bindings/power/allwinner,sun60i-a733-pck-600.h
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+
+#ifndef _DT_BINDINGS_POWER_SUN60I_A733_PCK600_H_
+#define _DT_BINDINGS_POWER_SUN60I_A733_PCK600_H_
+
+#define PD_VI			0
+#define PD_DE_SYS		1
+#define PD_VE_DEC		2
+#define PD_VE_ENC		3
+#define PD_NPU			4
+#define PD_GPU_TOP		5
+#define PD_GPU_CORE		6
+#define PD_PCIE			7
+#define PD_USB2			8
+#define PD_VO			9
+#define PD_VO1			10
+
+#endif /* _DT_BINDINGS_POWER_SUN60I_A733_PCK600_H_ */

From patchwork Thu Mar  5 03:34:08 2026
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Yuanshen Cao <alex.caoys@gmail.com>
X-Patchwork-Id: 14454004
Received: from mail-qt1-f169.google.com (mail-qt1-f169.google.com
 [209.85.160.169])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 66256264619
	for <linux-pm@vger.kernel.org>; Thu,  5 Mar 2026 03:34:14 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org;
 arc=none smtp.client-ip=209.85.160.169
ARC-Seal: i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1772681656; cv=none;
 b=DwPwq/mf/p3iMgqzszfCXY2b2bts3xiuFVMTC/Yc9gHXJnGuNWMV7WAw2dCEgyXMWsQSeNm8M4Ah2doklLZnr4BE4iDsoZcO0Lg/7I1RyHN/5tigR1AuQ18aWoFjegud6KrMq9rF5836J43/Nea5byDs6AYh3aOhjmSUgAFOrGE=
ARC-Message-Signature: i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1772681656; c=relaxed/simple;
	bh=3cQd24z258zjc20dy6ohb0HVeZL+LfFrMaR3/kR1p5Y=;
	h=From:Date:Subject:MIME-Version:Content-Type:Message-Id:References:
	 In-Reply-To:To:Cc;
 b=Asf1C/ZX8A005nREAW6Bspq8L4qjNS4GV2OEpeTq1UFk+Xh0H/lhGY30NQEj4JGOZyN2gkH/BI3iWkN2y5ON7Qj44feXB5UxBqMBofLy0tpIxYtyeamgbOvwVxrmem8SKmUWzTHdANEmYsGwO6pNMBYDc7rCchz8U/dDPSHZzO8=
ARC-Authentication-Results: i=1; smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=gmail.com;
 spf=pass smtp.mailfrom=gmail.com;
 dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com
 header.b=SidjP6Bj; arc=none smtp.client-ip=209.85.160.169
Authentication-Results: smtp.subspace.kernel.org;
 dmarc=pass (p=none dis=none) header.from=gmail.com
Authentication-Results: smtp.subspace.kernel.org;
 spf=pass smtp.mailfrom=gmail.com
Authentication-Results: smtp.subspace.kernel.org;
	dkim=pass (2048-bit key) header.d=gmail.com header.i=@gmail.com
 header.b="SidjP6Bj"
Received: by mail-qt1-f169.google.com with SMTP id
 d75a77b69052e-506989e8516so65352861cf.0
        for <linux-pm@vger.kernel.org>; Wed, 04 Mar 2026 19:34:14 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1772681653; x=1773286453;
 darn=vger.kernel.org;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:from:to:cc:subject:date:message-id
         :reply-to;
        bh=2d5yl02iJv/wkdm4hWnCdO+qs5aDrcJeSjvUW0fTJuE=;
        b=SidjP6Bj1PVb3X7Yjw+MI6l8teBl3jhBlDLyZuGxFtfUg7WFxijpnenVz19Qu1nQLX
         eBrVlTNn4oEpkPnaMyt5Lv8xmcZ3xmjgqNOiyyJovd/YcxxJZv3iNZnqKTi0+UJSXjzy
         ATKUsOyfK02qOUX7ZrBT3GsjrBVJ3wcy7JvZNGrCc6A/xIrmYdUA6ml/8eTKBRSGMfJ3
         onkEWzLMUAxnH8mll9PP69MFHxsQFfVbXgDfGpqWjG1RTqHaxy7OhqTEjLNSHcMashaf
         NkbQClKluWQ3zMw3+J2V1lA+57uk46CyZD/ZyORfWr6d/C3VyoC2lCBlyNRlqyh74lHC
         9rTg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1772681653; x=1773286453;
        h=cc:to:in-reply-to:references:message-id:content-transfer-encoding
         :mime-version:subject:date:from:x-gm-gg:x-gm-message-state:from:to
         :cc:subject:date:message-id:reply-to;
        bh=2d5yl02iJv/wkdm4hWnCdO+qs5aDrcJeSjvUW0fTJuE=;
        b=Jkbex1qiF/VSvjpRoyLBxaSszxNqalie1Nfht87ycIg5jBLeRDxZr/BCC/rYU5qdzW
         eCjRrSQMmmXtOnmUeyJ2tQwBrqh9qTlc4h9lMuXfYuOfg0uDy/L0WDSowlC2RnqSHsBW
         +SHmT7LU6Slut6nmXOrH2iBAC6byj5uZlLe/VcQmKV4JqUoUJFg0l/dqOxT9WyOmskjj
         1oIUzUxs1EvUQ8kIgNyW0ndKVj9QG7jNaLxkOB0T+ld4A5NzR+1+QFotS7zpsIEyZyXv
         A7gW7ueWYrFL5Oox5w+MPIRrj8wQDB9AcZEGSUihnlrX1MvB7khbjr90chFDLyOX+7yB
         L+jg==
X-Forwarded-Encrypted: i=1;
 AJvYcCUoIqsFv664CZXd4Xvf0tjNsKG8MeAWpqGTs142JqdyCyfEK3uj901QoJm9BOxDqfpv/WuDr5UGew==@vger.kernel.org
X-Gm-Message-State: AOJu0YwLaGIkpb2X2ll8xRHhm6yqkuJeH7fyATiZxIIaD+wejdncIGhf
	8GNU/jPd5e5ovZOXGaXU72MoJGul5BH3QrxNmyFKBPBVcWjMBHx21pD6
X-Gm-Gg: ATEYQzwHplXjUT1pEnW87LZGPICeZifIipIq7KNXl21Jw6tNpuNcFTngm6QN/seTDUi
	BzoP/k0G00egXn3urwbWdNKxLqCxpdwzFoXDkCt0XLkou7qm14nwobbMTvA8L7bKkXvuE44lt1Q
	U8L994HEsP2mY/BQp+aU9wpX7HArs7SXSYFLDC/P3GoXx08tnaLAmKLGNiWWYfGrrcmTcisq+PU
	8PtDiByqA+KEJji5EEJd78kce0MJ08Xzk+KInPUmQcZ9hnr6gaeyHWgmayt1FGub7LtFKXcOAWX
	8JJ1CLeAxHmLLuI1fqEO82bJSyIWlm7kGZ8tzad+sy+FwL8rfeuoGmfXhGO6uQo6aBby+X9HvE8
	rEOpFqCJpNZ16Noxzk73dv1G7/T0/LFifpZ8w0VIUHTnbEz6mvY/4yWIJWGnTeQ7eYZXj0wUrhA
	LPW/4oiMixNkuZvEV3keZmpqNyFj8=
X-Received: by 2002:ac8:5a0c:0:b0:501:4b10:aa9e with SMTP id
 d75a77b69052e-508db2a632bmr59254511cf.13.1772681653433;
        Wed, 04 Mar 2026 19:34:13 -0800 (PST)
Received: from [172.17.0.2] ([134.128.219.200])
        by smtp.gmail.com with ESMTPSA id
 d75a77b69052e-507449630b6sm190667601cf.7.2026.03.04.19.34.12
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 04 Mar 2026 19:34:13 -0800 (PST)
From: Yuanshen Cao <alex.caoys@gmail.com>
Date: Thu, 05 Mar 2026 03:34:08 +0000
Subject: [PATCH v2 2/2] pmdomain: sunxi: Add support for A733 to Allwinner
 PCK600 driver
Precedence: bulk
X-Mailing-List: linux-pm@vger.kernel.org
List-Id: <linux-pm.vger.kernel.org>
List-Subscribe: <mailto:linux-pm+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-pm+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Message-Id: <20260305-b4-pck600-a733-v2-2-ba6bbed7d253@gmail.com>
References: <20260305-b4-pck600-a733-v2-0-ba6bbed7d253@gmail.com>
In-Reply-To: <20260305-b4-pck600-a733-v2-0-ba6bbed7d253@gmail.com>
To: Rob Herring <robh@kernel.org>, Krzysztof Kozlowski <krzk+dt@kernel.org>,
 Conor Dooley <conor+dt@kernel.org>, Chen-Yu Tsai <wens@kernel.org>,
 Jernej Skrabec <jernej.skrabec@gmail.com>,
 Samuel Holland <samuel@sholland.org>, Ulf Hansson <ulf.hansson@linaro.org>
Cc: devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-sunxi@lists.linux.dev, linux-kernel@vger.kernel.org,
 linux-pm@vger.kernel.org, Yuanshen Cao <alex.caoys@gmail.com>
X-Mailer: b4 0.14.3

The Allwinner A733 PCK600, similar to A523 PCK600, is likely a
customized version of ARM PCK-600 power controller. It shares the same
BSP driver with A523. According to the BSP provided by Radxa, unlike
A523, it doesn't require reset, as well as a different pair of delay
values.

Make reset optional in the sunxi pck600 driver and add support
for A733.

Signed-off-by: Yuanshen Cao <alex.caoys@gmail.com>
Acked-by: Jernej Skrabec <jernej.skrabec@gmail.com>
Reviewed-by: Chen-Yu Tsai <wens@kernel.org> # matched against BSP driver
---
 drivers/pmdomain/sunxi/sun55i-pck600.c | 35 ++++++++++++++++++++++++++++++----
 1 file changed, 31 insertions(+), 4 deletions(-)

diff --git a/drivers/pmdomain/sunxi/sun55i-pck600.c b/drivers/pmdomain/sunxi/sun55i-pck600.c
index c7ab51514531..1d47bbd35ced 100644
--- a/drivers/pmdomain/sunxi/sun55i-pck600.c
+++ b/drivers/pmdomain/sunxi/sun55i-pck600.c
@@ -52,6 +52,7 @@ struct sunxi_pck600_desc {
 	u32 logic_power_switch0_delay;
 	u32 logic_power_switch1_delay;
 	u32 off2on_delay;
+	bool has_rst_clk;
 };
 
 struct sunxi_pck600_pd {
@@ -151,9 +152,11 @@ static int sunxi_pck600_probe(struct platform_device *pdev)
 	if (IS_ERR(base))
 		return PTR_ERR(base);
 
-	rst = devm_reset_control_get_exclusive_released(dev, NULL);
-	if (IS_ERR(rst))
-		return dev_err_probe(dev, PTR_ERR(rst), "failed to get reset control\n");
+	if (desc->has_rst_clk) {
+		rst = devm_reset_control_get_exclusive_released(dev, NULL);
+		if (IS_ERR(rst))
+			return dev_err_probe(dev, PTR_ERR(rst), "failed to get reset control\n");
+	}
 
 	clk = devm_clk_get_enabled(dev, NULL);
 	if (IS_ERR(clk))
@@ -206,7 +209,27 @@ static const struct sunxi_pck600_desc sun55i_a523_pck600_desc = {
 	.device_ctrl1_delay = 0xffff,
 	.logic_power_switch0_delay = 0x8080808,
 	.logic_power_switch1_delay = 0x808,
-	.off2on_delay = 0x8
+	.off2on_delay = 0x8,
+	.has_rst_clk = true,
+};
+
+static const char * const sun60i_a733_pck600_pd_names[] = {
+	"VI", "DE_SYS", "VE_DEC", "VE_ENC", "NPU",
+	"GPU_TOP", "GPU_CORE", "PCIE", "USB2", "VO", "VO1"
+};
+
+static const struct sunxi_pck600_desc sun60i_a733_pck600_desc = {
+	.pd_names = sun60i_a733_pck600_pd_names,
+	.num_domains = ARRAY_SIZE(sun60i_a733_pck600_pd_names),
+	.logic_power_switch0_delay_offset = 0xc00,
+	.logic_power_switch1_delay_offset = 0xc04,
+	.off2on_delay_offset = 0xc10,
+	.device_ctrl0_delay = 0x1f1f1f,
+	.device_ctrl1_delay = 0x1f1f,
+	.logic_power_switch0_delay = 0x8080808,
+	.logic_power_switch1_delay = 0x808,
+	.off2on_delay = 0x8,
+	.has_rst_clk = false,
 };
 
 static const struct of_device_id sunxi_pck600_of_match[] = {
@@ -214,6 +237,10 @@ static const struct of_device_id sunxi_pck600_of_match[] = {
 		.compatible	= "allwinner,sun55i-a523-pck-600",
 		.data		= &sun55i_a523_pck600_desc,
 	},
+	{
+		.compatible	= "allwinner,sun60i-a733-pck-600",
+		.data		= &sun60i_a733_pck600_desc,
+	},
 	{}
 };
 MODULE_DEVICE_TABLE(of, sunxi_pck600_of_match);
